src/metabase/sync/analyze/special_types/values.clj

```diff
@@ -0,0 +1,165 @@
+(ns metabase.sync.analyze.special-types.values
+  "Logic for inferring (and setting) the special types of fields based on tests done against a sequence of their values.
+   Also sets `:preview_display` to `false` if a Field has on average very long text values."
+  (:require [cheshire.core :as json]
+            [clojure.tools.logging :as log]
+            [metabase
+             [driver :as driver]
+             [util :as u]]
+            [metabase.db.metadata-queries :as queries]
+            [metabase.models
+             [field :refer [Field]]
+             [field-values :as field-values]]
+            [metabase.sync
+             [interface :as i]
+             [util :as sync-util]]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(def ^:private Values
+  "Schema for the VALUES passed to each of the functions below. *Guaranteed* to be non-nil and non-empty."
+  ;; Validating against this is actually pretty quick, in the order of microseconds even for a 10,000 value sequence
+  (s/constrained [(s/pred (complement nil?))] seq "Non-empty sequence of non-nil values."))
```

Comment:
we can definitely start saving that information down the road but I tried to keep the scope of this PR small so I skipped introducing new models that as of yet don't serve any purpose.

Don't want to fall into the RawTable/RawColumn trap :D